smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/InotifyEventApplier.java

```diff
@@ -80,31 +103,77 @@ public void apply(Event[] events) throws IOException, MetaStoreException {
     this.apply(Arrays.asList(events));
   }
 
+  private boolean shouldIgnore(String path) {
+    String toCheck = path.endsWith("/") ? path : path + "/";
+    for (String s : ignoreEventDirs) {
+      if (toCheck.startsWith(s)) {
+        return true;
+      }
+    }
+    if (fetchEventDirs.isEmpty()) {
+      return false;
+    }
+    for (String s : fetchEventDirs) {
+      if (toCheck.startsWith(s)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   private List<String> getSqlStatement(Event event) throws IOException, MetaStoreException {
+    String path;
+    String srcPath, dstPath;
+
     LOG.debug("Even Type = {}", event.getEventType().toString());
     switch (event.getEventType()) {
       case CREATE:
+        path = ((Event.CreateEvent) event).getPath();
+        if (shouldIgnore(path)) {
+          return Arrays.asList();
+        }
         LOG.trace("event type:" + event.getEventType().name() +
             ", path:" + ((Event.CreateEvent) event).getPath());
         return Arrays.asList(this.getCreateSql((Event.CreateEvent) event));
       case CLOSE:
+        path = ((Event.CloseEvent) event).getPath();
+        if (shouldIgnore(path)) {
+          return Arrays.asList();
+        }
         LOG.trace("event type:" + event.getEventType().name() +
             ", path:" + ((Event.CloseEvent) event).getPath());
         return Arrays.asList(this.getCloseSql((Event.CloseEvent) event));
       case RENAME:
+        srcPath = ((Event.RenameEvent) event).getSrcPath();
```

Comment:
If both src and dest are not ignored, the code are executed regularly. If both src and dest are ignored, return an empty list directly. If src is ignored, but dest is not ignored, insert the dest info to file table. If src is not ignored, but dest is ignored, delete the src info from file table.